google/cloud/bigquery/client.py

```diff
@@ -2174,7 +2174,15 @@ def load_table_from_dataframe(
         else:
             job_config = job.LoadJobConfig()
 
-        job_config.source_format = job.SourceFormat.PARQUET
+        if job_config.source_format:
+            if job_config.source_format is not job.SourceFormat.PARQUET:
```

Comment:
This (non)identity check is relying on interning strings, but we shouldn't rely on `"PARQUET"` strings to point (or not point) to the same string instance. Let's use the equality operator instead.

(`SourceFormat` is not an `enum`, it's just a plain class with a few string members)